File: packages/sdk-js/src/resources/customer_tools.ts
import { BaseResource } from './base';
import { AxiosRequestConfig } from 'axios';

// Per-tenant tool enablement (/api/customers/{customerId}/tools).

interface AddCustomerToolPayload {
  toolId: string;
  [key: string]: any;
}

export class CustomerToolsResource extends BaseResource {
  protected readonly RESOURCE_PATH = 'customers';

  /** Lists the tools configured for a customer. GET /api/customers/{customerId}/tools */
  async list(customerId: string, config?: AxiosRequestConfig): Promise<any[]> {
    const path = `${this.RESOURCE_PATH}/${customerId}/tools`;
    return this.httpClient.get<any[]>(path, config);
  }

  /** Adds a tool to a customer's configured tools. POST /api/customers/{customerId}/tools */
  async add(customerId: string, payload: AddCustomerToolPayload, config?: AxiosRequestConfig): Promise<any> {
    const path = `${this.RESOURCE_PATH}/${customerId}/tools`;
    return this.httpClient.post<any>(path, payload, config);
  }

  /** Removes a tool from a customer's configured tools. DELETE /api/customers/{customerId}/tools/{toolId} */
  async remove(customerId: string, toolId: string, config?: AxiosRequestConfig): Promise<any | null> {
    const path = `${this.RESOURCE_PATH}/${customerId}/tools/${toolId}`;
    return this.httpClient.delete<any>(path, config);
  }
}

File: packages/sdk-python/veltrix_sdk/resources/customer_tools.py
# -*- coding: utf-8 -*-

from .base import BaseResource


class CustomerToolsResource(BaseResource):
    """Per-tenant tool enablement (``/api/customers/{customerId}/tools``)."""
    RESOURCE_PATH = "customers"

    def list(self, customer_id, **kwargs):
        """Lists the tools configured for a customer.

        Corresponds to GET /api/customers/{customerId}/tools.
        """
        path = f"{self.RESOURCE_PATH}/{customer_id}/tools"
        return self._http_client.get(path, **kwargs)

    def add(self, customer_id, data, **kwargs):
        """Adds a tool to a customer's configured tools.

        Corresponds to POST /api/customers/{customerId}/tools.
        """
        path = f"{self.RESOURCE_PATH}/{customer_id}/tools"
        return self._http_client.post(path, data=data, **kwargs)

    def remove(self, customer_id, tool_id, **kwargs):
        """Removes a tool from a customer's configured tools.

        Corresponds to DELETE /api/customers/{customerId}/tools/{toolId}.
        """
        path = f"{self.RESOURCE_PATH}/{customer_id}/tools/{tool_id}"
        return self._http_client.delete(path, **kwargs)
